lib/flutter_inapp_purchase.dart

```diff
@@ -84,135 +86,121 @@ class FlutterInappPurchase {
   Future<String?> get initConnection async {
     if (_platform.isAndroid) {
       await _setPurchaseListener();
-      final String? result = await _channel.invokeMethod('initConnection');
-      return result;
+      return _channel.invokeMethod<String>('initConnection');
     } else if (_platform.isIOS) {
       await _setPurchaseListener();
-      final String? result = await _channel.invokeMethod('canMakePayments');
-      return result;
+      return _channel.invokeMethod<String>('canMakePayments');
     }
-    throw PlatformException(
-        code: _platform.operatingSystem, message: "platform not supported");
+
+    throw _platformException;
   }
 
   /// Retrieves a list of products from the store on `Android` and `iOS`.
   ///
   /// `iOS` also returns subscriptions.
   Future<List<IAPItem>> getProducts(List<String> skus) async {
-    skus = skus.toList();
+    if (skus.isEmpty) return [];
     if (_platform.isAndroid) {
-      dynamic result = await _channel.invokeMethod(
+      final result = await _channel.invokeListMethod<Map>(
         'getItemsByType',
         <String, dynamic>{
           'type': EnumUtil.getValueString(_TypeInApp.inapp),
           'skus': skus,
         },
       );
-      return extractItems(result);
+      return extractItems(result ?? []);
     } else if (_platform.isIOS) {
-      dynamic result = await _channel.invokeMethod(
+      final result = await _channel.invokeListMethod(
         'getItems',
-        {
-          'skus': skus,
-        },
+        {'skus': skus},
       );
-
-      return extractItems(json.encode(result));
+      return extractItems(result ?? []);
```

Comment:
Around the whole platform operations with strings were replaced with operations with maps. So method channels are sending maps mostly and there is no need to make another unnecessary json string encode/decode